go/mysql/conn.go

```diff
@@ -1279,6 +1298,124 @@ func (c *Conn) handleComPing() bool {
 	return true
 }
 
+// handleComQueryMulti is a newer version of handleComQuery that uses
+// the StreamExecuteMulti and ExecuteMulti RPC calls to push the splitting of statements
+// down to Vtgate.
+func (c *Conn) handleComQueryMulti(handler Handler, data []byte) (kontinue bool) {
+	c.startWriterBuffering()
+	defer func() {
+		if err := c.endWriterBuffering(); err != nil {
+			log.Errorf("conn %v: flush() failed: %v", c.ID(), err)
+			kontinue = false
+		}
+	}()
+
+	queryStart := time.Now()
+	query := c.parseComQuery(data)
+	c.recycleReadPacket()
+
+	res := c.execQueryMulti(query, handler)
+	if res != execSuccess {
+		return res != connErr
+	}
+
+	timings.Record(queryTimingKey, queryStart)
+	return true
+}
+
+// execQueryMulti is a newer version of execQuery that uses
+// the StreamExecuteMulti and ExecuteMulti RPC calls to push the splitting of statements
+// down to Vtgate.
+func (c *Conn) execQueryMulti(query string, handler Handler) execResult {
+	// lastPacketSent signifies whether we have sent the last packet to the client
+	// for a given query. This is used to determine whether we should send an
+	// end packet after the query is done or not. Initially we don't need to send an end packet
+	// so we initialize this value to true.
+	lastPacketSent := true
```

Comment:
nitpick: naming here is slightly confusing, since we start by claiming the last packet has been sent before anything has been sent . took me a couple of reads to understand it

Reply:
maybe we could call it `needsEndPacket` instead?

Reply:
Done